### PR TITLE
(SIMP-5974) Add support for incron 0.5.12+

### PR DESCRIPTION
**Pull request by trevor-vaughan**
- Add Incron::Mask Data Type denoting valid incron masks
- Added support for the new options starting with 0.5.12
  - Automatically strip out options that are not supported by earlier versions
    for seamless backward compatibility

SIMP-5974 #comment Update for handling incron 0.5.12+

### CODE REVIEW

File: lib/puppet/type/incron_system_table.rb [hunk 1]
```diff
@@ -44,7 +44,10 @@
         'IN_NO_LOOP',
         'IN_ONESHOT',
         'IN_ONLYDIR',
-        'IN_OPEN'
+        'IN_OPEN',
```
Comment (lnemsick-simp) [inline lib/puppet/type/incron_system_table.rb]: - Missing IN_MOVED_TO for latest version
- IN_NO_LOOP does not exist in latest version

Comment (trevor-vaughan) [reply lib/puppet/type/incron_system_table.rb]: @lnemsick-simp I think the omission of `IN_NO_LOOP` in the latest version is a mistake. At any rate, presence of the option did not cause any issues.

Added `IN_MOVED_TO` in a later PR, thanks!

File: lib/puppet/type/incron_system_table.rb [hunk 2]
```diff
@@ -66,7 +70,28 @@
     desc 'The incron "masks" to apply'
 
     munge do |value|
-      Array(value).join(',')
+      value = Array(value).join(',').split(',')
+
```
Comment (lnemsick-simp) [inline lib/puppet/type/incron_system_table.rb]: Wouldn't it be better to strip the unsupported features from valid_incron_masks, instead, so that invalid masks were not silently dropped?  As a user, I would be confused why my masks passed validation but didn't appear in the incron table.

File: CHANGELOG [hunk 3]
```diff
@@ -1,3 +1,10 @@
+* Sun Jan 20 2019 Trevor Vaughan <tvaughan@onyxpoint.com> - 0.4.0-0
+- Add the ability to set the 'max_open_files' ulimit
+- Add Incron::Mask Data Type denoting valid incron masks
+- Added support for the new options starting with 0.5.12
+  - Automatically strip out options that are not supported by earlier versions
+    for seamless backward compatibility
+
```
Comment (lnemsick-simp) [inline CHANGELOG]: Need to note new purge parameter

File: manifests/system_table.pp [hunk 4]
```diff
@@ -57,22 +60,24 @@
...
+  include 'incron'
 
   $_ensure = $enable ? { true => 'present', default => 'absent' }
 
   if $custom_content {
     incron_system_table { $name:
       ensure  => $_ensure,
-      content => $custom_content
+      content => $custom_content,
+      notify  => Class['incron::service']
```
Comment (lnemsick-simp) [inline manifests/system_table.pp]: Why is this needed? Per the incron man page:
>>If a table (incrontab) is changed incrond reacts immediately and reloads the table
 >> table. Currently running child processes (commands) are not affected.

File: lib/puppet/type/incron_system_table.rb (hunk 2 above)
Comment (trevor-vaughan) [reply lib/puppet/type/incron_system_table.rb]: @lnemsick-simp I don't know if that fact is available at that scope. I think that this is a reasonable improvement should probably be an additional ticket due to the importance of the fix.

File: manifests/system_table.pp (hunk 4 above)
Comment (trevor-vaughan) [reply manifests/system_table.pp]: @lnemsick-simp In testing, this did not always appear to occur.

**Comment by trevor-vaughan** [comment]
:hourglass_flowing_sand: https://gitlab.com/simp/pupmod-simp-incron/pipelines/44806299
